src/circshiftedarray.jl

```diff
@@ -45,44 +45,50 @@ CircShiftedVector(v::AbstractVector, n = (0,)) = CircShiftedArray(v, n)
 
 size(s::CircShiftedArray) = size(parent(s))
 
+
 @inline function bringwithin(idx::Int, range::AbstractRange)
-    a, b = extrema(range)
-    n = length(range)
-    while idx < a
-        idx += n
-    end
-    while idx > b
-        idx -= n
-    end
-    idx
+    t = mod(idx - first(range), last(range))
```

Comment:
This seems wrong for ranges with offsets (eg `2:10`). Maybe best to restrict to `AbstractUnitRange` and use `length(range)`.